src/codewhisperer/activation.ts

```diff
@@ -247,14 +247,13 @@ export async function activate(context: ExtContext): Promise<void> {
             const notificationLastShown =
                 context.extensionContext.globalState.get<Date>(
                     CodeWhispererConstants.accessTokenMigrationDoNotShowAgainLastShown
-                ) || t
+                ) || new Date()
```

Comment:
`Date` is serialized into a string by default. I would use the epoch time in milliseconds instead:
```ts
const currentTime = Date.now()
const lastShownTime = context.extensionContext.globalState.get<number>(
    CodeWhispererConstants.accessTokenMigrationDoNotShowAgainLastShown,
    currentTime
)
``` 
Then all math should be changed to work in milliseconds.

Reply:
> Then all math should be changed to work in milliseconds.

Since this is a hotfix should we wait on that as a followup?

Reply:
> Since this is a hotfix should we wait on that as a followup? (Great callout though)

No this is the root cause of the problem, not variable assignment order. I updated my comment to make it more clear.

Reply:
Thanks for this callout, will update

Reply:
Change the value of `accessTokenMigrationDoNotShowAgainLastShown` as well. Otherwise you'll have to handle state from 1.59.0

Reply:
const current_time = new Date()
const next_week =new Date(current_time.getTime() + 7 * 24 * 60 * 60 * 1000)

Then you can just compare the Date object

Reply:
Agree with what @JadenSimon said. Epoch time is better.

Reply:
Updated in latest commit